src/AWS/CRT/Auth/SigningConfigAWS.php

```diff
@@ -23,7 +23,8 @@ public static function defaults() {
             'signed_body_value' => null,
             'signed_body_header_type' => SignedBodyHeaderType::NONE,
             'expiration_in_seconds' => 0,
-            'date' => time()
+            'date' => time(),
+            'should_sign_header' => null,
```

Comment:
"should_sign_header" makes me think that this won't always sign it if it's set to true, is that correct?  Otherwise maybe  all it "enable_header_signing"

Reply:
It is never set to true, it's set to a callback that is invoked per-header to allow users to filter which headers get signed